### PR TITLE
Setup standard ECS GH workflows

### LINKED ISSUE #8: Setup GH Actions
## Expected Behavior
The repo has GH actions that actually run

## Current Behavior
This repo got set up with required GH actions, but they aren't set up to run for some reason. This is preventing PRs from being merged. 

## Possible Solution
Configure the GH actions to run.

### PR DESCRIPTION
**Pull request by slactjohnson**
## Description
Adds in standard ECS GH workflows. 

## Motivation and Context
Closes #8 

## How Has This Been Tested?
Will hopefully be tested when the workflows run on this PR....

### CODE REVIEW

**Comment by tangkong** [comment]
We transitioned to pyproject.toml for specifying test dependencies at some point.  Try adding something like this to `pyproject.toml`

https://github.com/pcdshub/atef/blob/11b15c01dfafd9fe3c6d5ea39afd5844c071c339/pyproject.toml#L39-L40

**Comment by ZLLentz** [comment]
Wow, that's a lot of commits and a well earned "all checks have passed"

**Comment by slactjohnson** [comment]
Thanks! I'd appreciate a once over if you can to see if there's anything weird or missing in the CI. It passes, but that doesn't mean it's correct...

File: .coveragerc [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [run]
 source =
-    btms_ui
+    btms-ui
```
Comment (tangkong) [inline .coveragerc:3]: Your experience here is why I dislike using dashes at all in package names

File: btms_ui/helpers.py [hunk 2]
```diff
@@ -34,8 +32,8 @@ def __init__(
         self,
         x: float = 0.0,
         y: float = 0.0,
-        channel_x: Optional[str] = None,
-        channel_y: Optional[str] = None,
+        channel_x: str | None = None,
```
Comment (tangkong) [inline btms_ui/helpers.py:35]: This syntax was introduced in py3.10, though the `from __future__ import annotations` lets you use it.  I tend to not use it to be safe, but it's fine here it seems

File: conda-recipe/meta.yaml [hunk 3]
```diff
@@ -1,38 +1,44 @@
-
-{% set data = load_setup_py_data() %}
-
+{% set package_name = "btms_ui" %}
```
Comment (tangkong) [inline conda-recipe/meta.yaml:1]: Now that both the package name and import name are `btms_ui`, we might consider renaming the repo and folder structure to use underscores to be consistent

**Review by tangkong — COMMENTED** [review]
I think this looks good on the whole, hard fought green checkmarks.  

The builds/tests seem to be running as intended.  Now it's just up to you to write tests that mean something.

File: btms_ui/__init__.py [hunk 4]
```diff
@@ -1,4 +1,3 @@
-from . import _version
-__version__ = _version.get_versions()['version']
+from ._version import __version__  # noqa: F401
```
Comment (ZLLentz) [inline btms_ui/__init__.py:1]: This might be missing when running interactively since it's generated by setuptools_scm later.

**Review by ZLLentz — COMMENTED** [review]
I didn't compare 1:1 with any other repo but to my eyes everything here looks correct aside from the thing that might `ImportError` for dev clones

File: pyproject.toml [hunk 5]
```diff
@@ -1,2 +1,39 @@
...
+[build-system]
+build-backend = "setuptools.build_meta"
+requires = [ "setuptools>=45", "setuptools_scm[toml]>=6.2",]
+
+[project]
+classifiers = [ "Programming Language :: Python :: 3",]
+description = "Beam Transport Motion System GUI and python code"
+dynamic = [ "version", "readme", "dependencies", "optional-dependencies",]
+keywords = []
+name = "btms-ui"
```
Comment (tangkong) [inline pyproject.toml:10]: This conflicts with the conda package name being `btms_ui`.  The builds confirm this:
* pip: `btms-ui                       0.1.dev135+g9befb1c.d20240717`
* conda: `btms_ui                   0.1.dev135+g9befb1c            py_0    local`

File: btms_ui/helpers.py (hunk 2 above)
Comment (ZLLentz) [reply btms_ui/helpers.py:35]: I think one of the linters asks you to change everything to this

Comment (tangkong) [reply btms_ui/helpers.py:35]: Interestingly none of the packages I work on do this yet 😆

File: btms_ui/__init__.py (hunk 4 above)
Comment (slactjohnson) [reply btms_ui/__init__.py:1]: I can't say I fully understand here, but it's working interactively on my clone. Once this is merged I can set up a clean clone and test that it still works interactively, and if not set up a new issue/PR to resolve that.

File: pyproject.toml (hunk 5 above)
Comment (slactjohnson) [reply pyproject.toml]: I've updated pyproject.toml to be consistent now

File: pyproject.toml [hunk 6]
```diff
@@ -1,2 +1,39 @@
...
+file = "LICENSE.md"
+
+[project.scripts]
+
+[tool.setuptools_scm]
+write_to = "btms_ui/_version.py"
+
+[tool.setuptools.dynamic.readme]
+file = "README.md"
+content-type = "text/markdown"
```
Comment (tangkong) [inline pyproject.toml:30]:
```suggestion
file = "README.rst"
```

The README file is actually a .rst (restructured text) file.  I believe setuptools takes rst as default so we can leave out the `content-type` key ([source](https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#dynamic-metadata))

**Review by tangkong — DISMISSED** [review]
I think this looks good, one last nitpick then I think we're good to merge this and tackle any followup bugs after.  

These migration diffs are always hard to parse.  Luckily the green check is a good indicator of success

**Comment by slactjohnson** [comment]
@tangkong I've committed your suggestion to fix README.rst, but now it requires a review again. Can you approve when you get a chance so I can merge?

**Review by tangkong — APPROVED** [review]

**Comment by slactjohnson** [comment]
Thank you!
